[Dylan Wong] iP

thewongdylan opened the pull request:
# DylanBot

> "Onions have layers. Ogres have layers... You get it? We both have layers." -Shrek ([Source](https://www.imdb.com/title/tt0126029/quotes/))

DylanBot is the last task tracking software you'll ever need. Some benefits include:
- Support for 3 different types of tasks (Todo, Deadline, Event)
- Singlish for local users
- Easy to use

To use DylanBot,
1. _Download_ it from this [link](https://github.com/thewongdylan/ip/releases/tag/A-Jar)
2. _Double click_ to open
3. _Add_ your tasks and let it track for you!

Not to mention, DylanBot is completely **FREE OF CHARGE**!
Wahoo 💯 💯 

DylanBot Features:
- [x] Tracking and managing tasks
- [x] Ability to search for keywords
- [ ] Sorting by time (coming soon)
- [ ] Reminders (coming soon)

For now, here's a cheeky sneak peek of the code that makes DylanBot tick, its `run` method:
```java
public void run() {
        ui.sendGreeting();
        Parser ps = new Parser(ui, tl);
        String command;
        try {
            while (!(command = ui.takeInput()).equals("bye")) {
                ps.process(command);
            }
        } catch (DylanBotException e) {
            ui.displayError(e);
        }
        try {
            st.saveDataToFile(tl);
        } catch (IOException e) {
            ui.displayIoError();
        }
        ui.sendExit();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

thewongdylan committed:
Level-0

Added greeting

thewongdylan committed:
Level-1

Added echo and bye

thewongdylan committed:
Level-2

Added list

thewongdylan committed:
Level-3

Added mark and unmark features

thewongdylan committed:
Level 4

Added task classification

thewongdylan committed:
A-TextUiTesting

thewongdylan committed:
Level 5

Added error handling

thewongdylan committed:
Level 6

Added delete feature

thewongdylan committed:
Level 6 Revisions

Added Task class and relevant subclasses

thewongdylan committed:
Revise Level 6

Fixed previously non-functional Exception throwing

thewongdylan committed:
Add Level 7

Added save/load through write/read from local data file

thewongdylan committed:
Add datetime support

Dates and times treated as proper DateTime rather than Strings

thewongdylan committed:
Merge branch 'branch-Level-8'
Added datetime support

thewongdylan committed:
Add more OOP

thewongdylan committed:
Add OOP classes

Abstracted Ui, Parser, Storage, TaskList classes

thewongdylan committed:
Merge branch 'A-MoreOOP'

thewongdylan committed:
Add Package dylanbot

thewongdylan committed:
Add Package dylanbot

thewongdylan committed:
Merge branch 'branch-A-Packages'

thewongdylan committed:
Merge branch 'add-gradle-support'

thewongdylan committed:
Add JUnit Tests

Added 2 simple JUnit Tests for basic functionality

thewongdylan committed:
Merge branch 'branch-A-JUnit'

thewongdylan committed:
Update build.gradle

Updated file path and file name of Jar file

thewongdylan committed:
Merge branch 'branch-A-Jar'

thewongdylan committed:
Add JavaDocs

Added JavaDocs for all necessary public methods

thewongdylan committed:
Add Find function

Added basic Find function for users to search for tasks based on a keyword

thewongdylan committed:
Enforce Java Coding Standard

Used checkstyle tests to enforce proper Java Coding Standard

thewongdylan committed:
Merge branch 'branch-Level-9'

thewongdylan committed:
Merge branch 'branch-A-CodingStandard'

thewongdylan committed:
Merge branch 'branch-A-JavaDoc'

thewongdylan committed:
Format code

Formatted code using checkstyle, added JavaDocs where needed

thewongdylan committed:
Merge branch 'branch-A-CheckStyle'

swtan346 commented on src/main/java/dylanbot/Task.java:
```diff
@@ -0,0 +1,57 @@
+
+    private String type;
+    private String desc;
+    private boolean completed;
```
Perhaps you can name it such that it sounds more boolean (e.g. isComplete) ?

swtan346 commented on src/main/java/dylanbot/DylanBot.java:
```diff
@@ -0,0 +1,58 @@
+ */
+public class DylanBot {
+    private Ui ui;
+    private Storage st;
+    private Parser ps;
+    private TaskList tl;
```
Maybe you can name them more explicitly instead of abbreviations?

swtan346 commented on src/main/java/dylanbot/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+     *
+     * @return whether the list of tasks is empty
+     */
+    public boolean isEmpty() {
```
Maybe you can name your method as a verb (e.g. checkEmpty) ?

swtan346 commented on src/main/java/dylanbot/DylanBot.java:
Maybe you can create more packages instead of everything in one package so that it's more organized.

swtan346 reviewed (commented):
Overall, good job. Just be more mindful when naming methods/variables.

thewongdylan committed:
Implement JavaFX Tutorial

Implemented basic GUI that echoes user input

thewongdylan committed:
Add GUI for DylanBot

Added basic functionality (transfer from text-based version), will improve design and iron out details (greeting msg) in future commits

thewongdylan committed:
Merge branch 'branch-Level-10'

AwJoey commented on src/main/java/dylanbot/Task.java:
```diff
@@ -0,0 +1,57 @@
+     * @param desc The specified description
+     */
+    public Task(String type, String desc) {
+        this.type = type;
```
Love that you have a type attribute so that you can change the value easily if needed! 💯

AwJoey commented on src/main/java/dylanbot/Task.java:
```diff
@@ -0,0 +1,57 @@
+/**
+ * Represents a general Task
+ */
+public class Task {
```
Perhaps you can make this abstract? I think this will prevents errors when implementing since Task does not have a type

AwJoey commented on src/main/java/dylanbot/EventTask.java:
```diff
@@ -0,0 +1,42 @@
+        return to;
+    }
+
+    @Override
```
I think you can add a description for the toString method here!

AwJoey commented on src/main/java/dylanbot/Storage.java:
```diff
@@ -0,0 +1,110 @@
+        String nextLine;
+        ArrayList<Task> res = new ArrayList<>();
+        try {
+            while ((nextLine = reader.readLine()) != null) {
```
Nice implementation! Perhaps you could extract some of this code and put in Parser class instead?

AwJoey reviewed (commented):
Nice work!! 💯 Just some suggestions that might make your code look even more polished 👍

thewongdylan committed:
Modify variable and method names

Edited boolean variables to reflect boolean properties

Edited boolean getter functions to reflect purpose

thewongdylan committed:
Merge branch 'branch-Level-10'

thewongdylan committed:
Improve Code Quality

Removed dead code
Reduced length of long methods, employed SLAP

thewongdylan committed:
Add JavaDocs

Added JavaDocs for refactored methods

thewongdylan committed:
Add assertions

Added assert statements to verify values at critical points of code

thewongdylan committed:
Merge pull request #2 from thewongdylan/branch-A-Assertions

Add assertions

thewongdylan committed:
Merge branch 'master' into branch-A-CodeQuality

thewongdylan committed:
Merge pull request #3 from thewongdylan/branch-A-CodeQuality

Improve Code Quality

thewongdylan committed:
Fix functionality

Rectified incorrect parsing of user input
Edited Task to be abstract (code quality)

thewongdylan committed:
Add tagging function

Added ability to tag tasks

Tasks can be tagged with multiple tags

tagList is written to a different file from taskList

Users can filter by tasks by tag

thewongdylan committed:
Merge branch 'branch-C-Tagging'

thewongdylan committed:
Improve GUI

Improved look of GUI
Added more interactions with DylanBot (greeting and exit)

For future improvements:
- Resizeability of window

thewongdylan committed:
Merge branch 'branch-A-BetterGui'

thewongdylan committed:
Update User Guide

Updated user guide
Added Ui.png for sample image of DylanBot in use

Added minor JavaDocs and cleaned up minor checkstyle violations

thewongdylan committed:
Add anchor links for sections

Added anchor links for all sections in the user guide

thewongdylan committed:
Fix typo

Minor typo fix

thewongdylan committed:
Merge branch 'branch-A-UserGuide'

thewongdylan committed:
Publish DylanBot v0.2

Published jar file to GitHub

thewongdylan committed:
Merge branch 'branch-A-Release'

thewongdylan committed:
Modify JAR file

Included main() method that allowed JAR file to function

Should pass smoke test now

thewongdylan committed:
Merge branch 'branch-A-Release'
